File: app/src/main/java/com/example/sachinsandbhor/retrofitsample/presentation/movieList/MovieListPresenter.java
package com.example.sachinsandbhor.retrofitsample.presentation.movieList;

import com.example.sachinsandbhor.retrofitsample.model.MovieResponse;

/**
 * Created by Sachin.Sandbhor on 05-02-2018.
 */

public class MovieListPresenter implements MovieListContract.MovieListPresenter, MovieListContract.MovieListModel.MovieListPresenterCallBack {

    MovieListContract.MovieListView movieListView;
    MovieListContract.MovieListModel movieListModel;

    public MovieListPresenter(MovieListContract.MovieListView movieListView) {
        this.movieListView = movieListView;
        movieListModel = new MovieListModel(this);
    }

    @Override
    public void getMovieList() {
        movieListView.startLoadingPost();
        movieListModel.getMovieList();
    }

    @Override
    public void onMovieListReceived(MovieResponse movieResponse) {
        movieListView.stopLoadingPost();
        if(movieListView != null){
            movieListView.setMovieList(movieResponse);
        }
    }

    @Override
    public void navigateToMovieDetail(int id) {
        if(null != movieListView) {
            movieListView.navigateToMovieDetail(id);
        }
    }

    @Override
    public void detach() {
        movieListView = null;
    }

    @Override
    public void onError(String errorMessage) {
        if(null != movieListView) {
            movieListView.onError(errorMessage);
        }
    }

    @Override
    public void setView(MovieListContract.MovieListView movieListView) {
        this.movieListView = movieListView;
    }
}

File: retrofitkotlin/src/main/java/com/sachinsandbhor/retrofitkotlin/presentation/movielist/MovieListPresenter.kt
package com.sachinsandbhor.retrofitkotlin.presentation.movielist

import com.sachinsandbhor.retrofitkotlin.domain.MovieListResponse


/**
 * Created by Sachin.Sandbhor on 08-03-2018.
 */
class MovieListPresenter(private var view: MovieListContract.MovieListView) :MovieListContract.MovieListPresenter,MovieListContract.MovieListModel.MovieListCallback {
    override fun navigateUser(id: Int) {
        view.navigateUser(id)
    }


    override fun setView(movieListView: MovieListContract.MovieListView) {
        view = movieListView
    }

    override fun getMovieList(pageNo : Int) {
        var movieListModel:MovieListContract.MovieListModel = MovieListModel(this)
        movieListModel.getMovieList(pageNo)
    }

    override fun movieListReceived(movieResponse: MovieListResponse) {
       view.setMovie(movieResponse)
    }

    override fun onError(message: String?) {
        view.onError(message)
    }
}